src/dotty/tools/dotc/typer/Checking.scala

```diff
@@ -156,16 +156,24 @@ object Checking {
         tp
     }
 
-    def apply(tp: Type) = tp match {
+    private def apply(tp: Type, cycleOK: Boolean, nestedCycleOK: Boolean): Type = {
```

Comment:
Maybe call these parameters `overrideCycleOK` or `cycleOKOverride` to avoid confusion with the class fields